app/components/MintBTC.tsx

```diff
@@ -4,15 +4,75 @@ import { Link } from "react-router";
 import { Button } from "./ui/button";
 import { FormProvider, useForm } from "react-hook-form";
 import { FormInput } from "./form/FormInput";
-import { useXverseWallet } from "./Wallet/XverseWallet/useWallet";
-import { useContext } from "react";
+import { useXverseConnect, useXverseWallet } from "./Wallet/XverseWallet/useWallet";
+import { useContext, useEffect, useState } from "react";
 import { WalletContext } from "~/providers/ByieldWalletProvider";
 import { Wallets } from "~/components/Wallet";
 import { FormNumericInput } from "./form/FormNumericInput";
 import { NumericFormat } from "react-number-format";
-import { formatBTC, parseBTC } from "~/lib/denoms";
+import { BTC, formatBTC, parseBTC } from "~/lib/denoms";
+import { nBTCMintTx } from "~/lib/nbtc";
+import { useToast } from "~/hooks/use-toast";
+import { Modal } from "./ui/dialog";
+import { Check } from "lucide-react";
+import { classNames } from "~/util/tailwind";
+import { isValidSuiAddress } from "@mysten/sui/utils";
+import { isMainNetNetwork } from "~/lib/appenv";
+import mainnetConfig from "~/config/mint/contracts-mainnet.json";
+import testnetConfig from "~/config/mint/contracts-testnet.json";
 
-const nBTCMintFeeInSatoshi = 10n;
+interface TransactionStatusProps {
+	SuiAddress: string;
+	txId: string;
+	handleRetry: () => void;
+}
+
+function TransactionStatus({ SuiAddress, txId, handleRetry }: TransactionStatusProps) {
+	const isMainNetMode = isMainNetNetwork();
+	// TODO: have one source of truth to get network details
```

Comment:
add this TODO to task

Reply:
Done:https://github.com/gonative-cc/byield/issues/320